### PR TITLE
Added Setup Guide for Ubuntu

### PR DESCRIPTION
**Pull request by PragatiVerma18**
- Added setup guide for Ubuntu 
- Added detailed instructions along with screenshots to make it easier to follow along
- Added tutorials and resources to set up the pre-requisites for the installation
- Added probable errors and their solutions

### CODE REVIEW

**Comment by tachyons** [comment]
Good one, wat about moving docs to `docs/` ?

**Comment by PragatiVerma18** [comment]
> Good one, wat about moving docs to `docs/`?

Okay, as you say. @tachyons Please let me know what would be best to do.

**Comment by satu0king** [comment]
@PragatiVerma18 Thanks for this! We needed a detailed documentation

**Comment by satu0king** [comment]
@PragatiVerma18 we actually have documentation scattered around. Could you help consolidate the same? 
We have documentation at 

https://github.com/CircuitVerse/CircuitVerse/blob/master/SETUP.md
https://github.com/CircuitVerse/CircuitVerse/wiki/Create-Apps
https://github.com/CircuitVerse/CircuitVerse/wiki/Development-on-Gitpod
https://github.com/CircuitVerse/CircuitVerse/wiki/Pull-Requests-using-Gitpod
https://github.com/CircuitVerse/CircuitVerse/wiki/Running-the-server
https://github.com/CircuitVerse/CircuitVerse/wiki/Preview-mail-in-development-environment-:-Mailcatcher

**Comment by abstrekt** [comment]
> @PragatiVerma18 we actually have documentation scattered around. Could you help consolidate the same?
> We have documentation at
> 
> https://github.com/CircuitVerse/CircuitVerse/blob/master/SETUP.md
> https://github.com/CircuitVerse/CircuitVerse/wiki/Create-Apps
> https://github.com/CircuitVerse/CircuitVerse/wiki/Development-on-Gitpod
> https://github.com/CircuitVerse/CircuitVerse/wiki/Pull-Requests-using-Gitpod
> https://github.com/CircuitVerse/CircuitVerse/wiki/Running-the-server
> https://github.com/CircuitVerse/CircuitVerse/wiki/Preview-mail-in-development-environment-:-Mailcatcher
> 
> I recommend moving everything to `/docs`. Because the wiki can be edited by anyone and it doesn't allow us to do PR - review properly.
> 
> @Shivansh2407 any inputs on this?
> @aayush-05, @abstrekt can you help with windows setup here.

surely.

**Comment by PragatiVerma18** [comment]
> > @PragatiVerma18 we actually have documentation scattered around. Could you help consolidate the same?
> > We have documentation at
> > https://github.com/CircuitVerse/CircuitVerse/blob/master/SETUP.md
> > https://github.com/CircuitVerse/CircuitVerse/wiki/Create-Apps
> > https://github.com/CircuitVerse/CircuitVerse/wiki/Development-on-Gitpod
> > https://github.com/CircuitVerse/CircuitVerse/wiki/Pull-Requests-using-Gitpod
> > https://github.com/CircuitVerse/CircuitVerse/wiki/Running-the-server
> > https://github.com/CircuitVerse/CircuitVerse/wiki/Preview-mail-in-development-environment-:-Mailcatcher
> > I recommend moving everything to `/docs`. Because the wiki can be edited by anyone and it doesn't allow us to do PR - review properly.
> > @Shivansh2407 any inputs on this?
> > @aayush-05, @abstrekt can you help with windows setup here.
> 
> surely.

@satu0king thanks for the review, I will do it. Just allow me some time.

File: SETUP IN UBUNTU.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+error @rails/webpacker@4.2.2: The engine "node" is incompatible with this module. Expected version ">=8.16.0". Got "8.10.0"
+error Found incompatible module.
+```
+then, try this:
+```
+$ rm yarn.lock
+$ bundle
+$ sudo npm cache clean -f
+$ sudo npm install -g n
+$ sudo n stable
```
Comment (tachyons) [inline SETUP IN UBUNTU.md:107]: What is `n` here ? local alias ?

File: SETUP IN UBUNTU.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+$ sudo apt-get install --reinstall nodejs-legacy
```
Comment (tachyons) [inline SETUP IN UBUNTU.md:108]: Prefer `nvm` for installing node

File: SETUP IN UBUNTU.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+The admin credentials after seeding will be:
+```
+User: Admin
+Email: admin@circuitverse.org
+Password: password
+```
+## Step 8 — :tada: Run Rails Server
+To run the rails server:
+```
+$ rails s -b 127.0.0.1 -p 8080
```
Comment (tachyons) [inline SETUP IN UBUNTU.md:192]: just `rails s` should be enough for most of the usecases

**Comment by tachyons** [comment]
Closing due to inactivity, feel free to reopen if you wish to continue working on this.

**Pull request closed without merging**